File: FSharp/AdventOfCode2.fs
let area (l, w, h) = 
   let min  = [l; w; h] |> List.sort |> List.take(2) |> List.fold(fun acc x -> acc * x) 1
   2 * l * w + 2 * w * h + 2 * h * l + min

[<EntryPoint>]
let main args = 
    let res = area(1,1,10)
    printf "%A" res
    0

File: Elixir/AdventOfCode2.ex
defmodule AdventOfCode2 do
  defp area(l, w, h) do
    min = [l, w, h] |> Enum.sort |> Enum.take(2) |> Enum.reduce(fn acc, x -> acc * x end)
    (2* (l * w)) + (2 * (w * h)) + (2 * (l * h)) + min
  end
  def solve(l, w, h) do
    area(l, w, h)
  end
end
